src/main/java/com/ticketbooking/repo/TicketRepo.java

```diff
@@ -0,0 +1,19 @@
+package com.ticketbooking.repo;
+
+import com.ticketbooking.model.Ticket;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+
+@Repository
+public interface TicketRepo extends JpaRepository<Ticket, Long> {
+
+    @Query("select t.seat.id from Ticket t join t.reservation r join t.screening s " +
+            "where t.screening.id = :screeningId " +
+            "and (r.confirmed = true " +
+            "or (r.reservationTime between (SYSTIMESTAMP - 0.010416) AND SYSTIMESTAMP and SYSTIMESTAMP < s.screeningTime - 0.010416))")
```

Comment:
Co to za magiczna wartość?

Reply:
Tak zgadza się, mi się też nie podoba. ona odzwierciedla wartość 15 minut, próba zapisania jej w postaci 15/1440 tutaj nie zadziałała.